### PR TITLE
feat: add magic trait to call get_canister

### PR DESCRIPTION
**Pull request by fcavazzoli**
### Feat
 - Add `get_canister` call to magic_bridge interface.
 - Ask for canister_id to magic_bridge for a given eth_address (as principal) when calling `dip20proxy.burn`

### CODE REVIEW

File: magic_bridge/ic/src/dip20_proxy/src/common/magic.rs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+                    return Err(TxError::Other(format!(
+                        "RejectionCode: {:?}\n{}",
+                        code, err
+                    )))
+                }
+            };
+
+        match get_canister {
+            (Ok(Some(canister_id)),) => Ok(canister_id),
+            (Ok(None),) => Err(TxError::Other(format!(
```
Comment (b0xtch) [inline magic_bridge/ic/src/dip20_proxy/src/common/magic.rs:28]: we could perhaps combine the `error` and `none`

**Review by b0xtch — APPROVED** [review]

File: magic_bridge/ic/src/dip20_proxy/src/common/magic.rs (hunk 1 above)
Comment (fcavazzoli) [reply magic_bridge/ic/src/dip20_proxy/src/common/magic.rs:28]: I like the idea of having them separated so we could differentiate an error and a not existing canister

Comment (b0xtch) [reply magic_bridge/ic/src/dip20_proxy/src/common/magic.rs:28]: But the method returns a result of either and ok or err no?

Comment (fcavazzoli) [reply magic_bridge/ic/src/dip20_proxy/src/common/magic.rs]: fixed 👍

**Review by b0xtch — APPROVED** [review]

**Pull request merged**